### PR TITLE
Make spec pulls more robust

### PR DESCRIPTION
**Pull request by jayrbolton**
Use gzipped github releases and the github API to pull spec updates.

Added a sanity check test case and some minimal docs

Related issue https://github.com/kbase/relation_engine_api/issues/10

### CODE REVIEW

File: src/relation_engine_server/api.py [hunk 1]
```diff
@@ -65,13 +65,17 @@ def show_view(name):
...
-@api.route('/refresh_specs', methods=['GET'])
+@api.route('/update_specs', methods=['GET'])
 def refresh_specs():
     """
-    Manually pull from the spec git repo to get updates.
+    Manually check for updates, download spec releases, and init new collections.
     """
-    git_output = pull_spec.pull_spec()
-    return flask.jsonify({"updates": git_output})
+    auth.require_auth_token(['RE_ADMIN'])
```
Comment (JamesJeffryes) [inline src/relation_engine_server/api.py]: Good call.

File: src/relation_engine_server/pull_spec.py [hunk 2]
```diff
@@ -1,21 +1,91 @@
...
+    # Download and extract a new release to /spec/repo
+    spec_repo_path = os.path.join(_spec_dir, 'repo')
+    tarball_url = release_info['tarball_url']
+    resp = requests.get(tarball_url, stream=True)
+    temp_file = tempfile.NamedTemporaryFile()
+    # Download from the tarball url to the temp file
+    _download_file(resp, temp_file.name)
+    # Extract the downloaded tarball into the spec path
+    _extract_tarball(temp_file.name, _spec_dir)
+    temp_file.close()  # Also deletes it
```
Comment (JamesJeffryes) [inline src/relation_engine_server/pull_spec.py]: Why not just use a context manager?

File: src/relation_engine_server/api.py (hunk 1 above)
Comment (JamesJeffryes) [reply src/relation_engine_server/api.py]: Although you'll need to add a service token (long lasting) to the cron job that calls this

File: src/relation_engine_server/pull_spec.py (hunk 2 above)
Comment (jayrbolton) [reply src/relation_engine_server/pull_spec.py]: Just wanted to reduce nesting for better or worse.. but i'll add a context manager here

**Comment by jayrbolton** [comment]
Regarding pulling prereleases for CI, we could make a PR later on. (agile!!) Made an issue for it here: https://github.com/kbase/relation_engine_api/issues/13

My plan was to hit the https://api.github.com/repos/kbase/relation_engine_api/releases endpoint to get the latest prerelease.

**Pull request merged**